src/utils/helpers/application/buildQueryFilters.ts

```diff
@@ -0,0 +1,119 @@
+import { DateTime } from 'luxon'
+import { QueryObject } from '../../hooks/useRouter'
+import { ApplicationStatus, ApplicationOutcome } from '../../../utils/generated/graphql'
+
+export default function buildQueryFilters(filters: QueryObject) {
+  const graphQLfilter = Object.entries(filters).reduce((filterObj, [key, value]) => {
+    if (!mapQueryToFilterField?.[key]) return filterObj
+    return { ...filterObj, ...mapQueryToFilterField[key](value) }
+  }, {})
+  return graphQLfilter
+}
+interface FilterMap {
+  [key: string]: (value: string) => object
+}
+
+const mapQueryToFilterField: FilterMap = {
+  type: (value: string) => {
+    return { templateCode: { equalToInsensitive: value } }
+  },
+  // category -- not yet implemented in schema
+  stage: (values: string) => {
+    return {
+      stage: inList(values),
+    }
+  },
+  status: (values: string) => {
+    return { status: inEnumList(values, ApplicationStatus) }
+  },
+  outcome: (values: string) => {
+    return { outcome: inEnumList(values, ApplicationOutcome) }
+  },
+  // action
+  // assigned
+  // consolidator
+  applicant: (values: string) => {
+    return {
+      or: [
+        { applicant: inList(values) },
+        { applicantFirstName: inList(values) },
+        { applicantLastName: inList(values) },
+        { applicantUsername: inList(values) },
+      ],
+    }
+  },
+  org: (values: string) => {
+    return { orgName: inList(values) }
+  },
+  lastActiveDate: (value: string) => {
+    const [startDate, endDate] = parseDateString(value)
+    console.log('Dates:', startDate, endDate)
+    return { lastActiveDate: { greaterThanOrEqualTo: startDate, lessThan: endDate } }
+  },
+  // deadlineDate (TBD)
+  search: (value: string) => {
+    return {
+      or: [
+        { name: { includesInsensitive: value } },
+        { applicantUsername: { includesInsensitive: value } },
+        { applicant: { includesInsensitive: value } },
+        { orgName: { includesInsensitive: value } },
+        { templateName: { includesInsensitive: value } },
+        { stage: { startsWithInsensitive: value } },
+      ],
+    }
+  },
+}
+
+const splitCommaList = (values: string) => values.split(',')
+
+// Use this if the values can be free text strings (e.g. stage name)
+const inList = (values: string) => {
+  return { inInsensitive: splitCommaList(values) }
+}
+
+// Use this if the values must conform to an Enum type (e.g. status, outcome)
+const inEnumList = (values: string, enumList: any) => {
+  return {
+    in: splitCommaList(values)
+      .map((value) => value.toUpperCase())
+      .filter((value) => {
+        return Object.values(enumList).includes(value)
+      }),
+  }
+}
+
+const parseDateString = (dateString: string) => {
+  if (mapNamedDates?.[dateString]) return mapNamedDates[dateString]
```

Comment:
Why not using `mapNamedDates[dateString]`? I'm confused what was the problem.

Reply:
Yeah I can see how that is confusing to read. What it's doing is, if the `dateString` is one of the "named" dates (e.g. "today"), just return the converted date value from the mapped object. If not, do some further processing and checks.

I've changed it to this, which is much clearer, I think:
```if (dateString in mapNamedDates) return mapNamedDates[dateString]```